cmake/OpenSSL.cmake

```diff
@@ -1,15 +1,34 @@
+# Check if a file is a symbolic link (macOS)
+function(is_symlink filepath result)
+    execute_process(
+        COMMAND "test" "-h" "${filepath}"
+        RESULT_VARIABLE return_code
+    )
+    if(return_code EQUAL 0)
+        set(${result} TRUE PARENT_SCOPE)
+    else()
+        set(${result} FALSE PARENT_SCOPE)
+    endif()
+endfunction()
+
 if(NOT DEFINED OPENSSL_CONFIGURED)
     if(APPLE)
-        execute_process(COMMAND brew --prefix openssl@3.1
-                OUTPUT_VARIABLE OPENSSL_PREIX RESULT_VARIABLE OPENSSL_RC
+        execute_process(COMMAND brew --prefix openssl@3
+                OUTPUT_VARIABLE OPENSSL_PREFIX RESULT_VARIABLE OPENSSL_RC
                 OUTPUT_STRIP_TRAILING_WHITESPACE)
 
         if(NOT OPENSSL_RC EQUAL 0)
             message(FATAL_ERROR "No OpenSSL found, use homebrew to install one")
         endif()
-
-        message(STATUS "Found OpenSSL at ${OPENSSL_PREIX}")
-        set(OPENSSL_ROOT_DIR ${OPENSSL_PREIX} CACHE INTERNAL "OpenSSL")
+        is_symlink(${OPENSSL_PREFIX} is_symlink_result)
+        message(STATUS "Found OpenSSL at ${OPENSSL_PREFIX}")
+        if(is_symlink_result)
```

Comment:
cmake has native "is_symlink" functionality: https://cmake.org/cmake/help/latest/command/if.html#is-symlink

Reply:
Also, thinking of this a bit deeper, it might just make sense always to get the absolute path, whether it is a symlink or not, making the code a lot simpler.